recipes/debug_assert/0001-Remove-DEBUG_ASSERT_PURE_FUNCTION.patch

```diff
@@ -0,0 +1,62 @@
+From 130adcbb393befa29d70036d93905f7bd94aa110 Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Jonathan=20M=C3=BCller?= <git@foonathan.net>
+Date: Thu, 13 Jun 2019 09:18:49 +0200
+Subject: [PATCH] Remove DEBUG_ASSERT_PURE_FUNCTION
+
+Fixes #17.
+---
+ README.md        |  2 --
+ debug_assert.hpp | 15 +++------------
+ 2 files changed, 3 insertions(+), 14 deletions(-)
+
+diff --git a/README.md b/README.md
+index b6af99e..e99fe19 100644
+--- a/README.md
++++ b/README.md
+@@ -54,8 +54,6 @@ The options are named like the macros.
+
+ <pre><code class="language-cpp">#define <a href='doc_debug_assert.md#debug_assert.hpp'>DEBUG_ASSERT_MARK_UNREACHABLE</a>
+
+-#define <a href='doc_debug_assert.md#debug_assert.hpp'>DEBUG_ASSERT_PURE_FUNCTION</a>
+-
+ #define <a href='doc_debug_assert.md#debug_assert.hpp'>DEBUG_ASSERT_FORCE_INLINE</a>
+
+ #define <a href='doc_debug_assert.md#DEBUG_ASSERT_CUR_SOURCE_LOCATION'>DEBUG_ASSERT_CUR_SOURCE_LOCATION</a>
+diff --git a/debug_assert.hpp b/debug_assert.hpp
+index 787d5d7..feaaa9d 100644
+--- a/debug_assert.hpp
++++ b/debug_assert.hpp
+@@ -44,15 +44,6 @@
+ #endif
+ #endif
+
+-#ifndef DEBUG_ASSERT_PURE_FUNCTION
```

Comment:
Why does a header-only library need source patching?

Reply:
There's a compilation problem with recent versions of GCC, it's noted in the recipe:
```
    # Patch from upstream that fixes compilation with recent versions of GCC
    - 0001-Remove-DEBUG_ASSERT_PURE_FUNCTION.patch
```

The problem is fixed upstream but there is no release with this included yet.

Reply:
OK, but that means this isn't version 1.3.3 of debug_assert, it is a prerelease or dev version because it contains some of the code that is already marked for a future version. Please ask upstream to cut a new release.